### PR TITLE
bs-platform: fix build on aarch64

### PR DESCRIPTION
**Pull request by anmonteiro**
###### Motivation for this change
https://github.com/BuckleScript/bucklescript/pull/4139

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @turboMaCk

### CODE REVIEW

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix
Comment (anmonteiro) [inline pkgs/development/compilers/bs-platform/build-bs-platform.nix]: I tracked down the issue that @bobzhang referred to in [this comment](https://github.com/NixOS/nixpkgs/pull/77726#discussion_r373833252).

Apparently running `node scripts/ninja.js config` will cause the build script to build the development version.

**Comment by anmonteiro** [comment]
Looks like we can build bs-platform on aarch64 again. Thanks for your help @bobzhang!

**Review by jonringer — COMMENTED** [review]
commit msg should be:
```
bs-platform: 7.0.1 -> 7.1.0
```

**Comment by anmonteiro** [comment]
Thanks, updated

File: pkgs/development/compilers/bs-platform/default.nix
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/default.nix]: this fails for me:

```
fatal: reference is not a tree: a7f482243fabb38292b81b7bae9cefcb8076237f
Unable to checkout a7f482243fabb38292b81b7bae9cefcb8076237f from https://github.com/BuckleScript/bucklescript.git.
```

**Comment by turboMaCk** [comment]
@anmonteiro 7.1.0 seems not to be released. Further more git commits of development version are changing so referencing those commits is not safe. We need to wait for the official release of 7.1.0 and commit that will stay in the master.

**Review by turboMaCk — CHANGES_REQUESTED** [review]
wait for 7.1.0 release

File: pkgs/development/compilers/bs-platform/default.nix
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/default.nix]: revert version bump:
```suggestion
  version = "7.0.1";
```

File: pkgs/development/compilers/bs-platform/default.nix
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/default.nix]: revert to old commit:

```suggestion
    rev = "52770839e293ade2bcf187f2639000ca0a9a1d46";
```

File: pkgs/development/compilers/bs-platform/default.nix
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/default.nix]: revert checksum:
```suggestion
    sha256 = "0s7g2zfhshsilv9zyp0246bypg34d294z27alpwz03ws9608yr7k";
```

**Review by turboMaCk — CHANGES_REQUESTED** [review]
I recommend leaving bump of version out from this and merging it **as a fix for arm build**. If you chose to do so please change the commit message. @jonringer 's suggested name follows convention in case of updates but not if we leave it out.

**Review by turboMaCk — APPROVED** [review]
LGTM thanks

**Review by turboMaCk — APPROVED** [review]

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix
Comment (jonringer) [inline pkgs/development/compilers/bs-platform/build-bs-platform.nix]: do you mind leaving a comment, so we don't stumble upon this again
```suggestion
  # avoid building the development version, will break aarch64 build
  dontConfigure = true;
```

**Comment by turboMaCk** [comment]
@anmonteiro can you please rebase and sqush commits in here? It seems to me that once that's done we can start pushing for merge.

**Comment by anmonteiro** [comment]
Squashed!

**Comment by marsam** [comment]
Thanks!

**Pull request merged**
